util/src/test/java/io/zeebe/util/health/DelayedHealthIndicatorTest.java

```diff
@@ -0,0 +1,170 @@
+            () -> new DelayedHealthIndicator(mockHealthIndicator, Duration.ofMillis(-50)))
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void shouldReportUnknownHealthStatusIfAskedBeforeDelegateHealthIndicatorWasCalled() {
+    // given
+    final var sutDelayedHealthIndicator =
+        new DelayedHealthIndicator(mockHealthIndicator, TEST_MAX_DOWNTIME);
+
+    // when
+    final Health actualHealth = sutDelayedHealthIndicator.health();
+
+    // then
+    Assertions.assertThat(actualHealth).isNotNull();
+    Assertions.assertThat(actualHealth.getStatus()).isEqualTo(Status.UNKNOWN);
+  }
+
+  @Test
+  public void
+      shouldReportHealthStatusOfDelegateHealthIndicatorIfBackendHealthIndicatorWasNeverUp() {
+    // given
+    final var sutDelayedHealthIndicator =
+        new DelayedHealthIndicator(mockHealthIndicator, TEST_MAX_DOWNTIME);
+
+    Mockito.when(mockHealthIndicator.health()).thenReturn(Health.down().build());
+
+    // when
+    sutDelayedHealthIndicator.checkHealth();
+    final Health actualHealth = sutDelayedHealthIndicator.health();
+
+    // then
+    Assertions.assertThat(actualHealth).isNotNull();
+    Assertions.assertThat(actualHealth.getStatus()).isEqualTo(Status.DOWN);
+  }
+
+  @Test
+  public void
+      shouldReportHealthStatusUpWhenBackendHealthIndicatorWasUpInThePastAndIsTemporarilyDown() {
+    // given
+    final var sutDelayedHealthIndicator =
+        new DelayedHealthIndicator(mockHealthIndicator, TEST_MAX_DOWNTIME);
+    // backend health indicator was up in the past
+    Mockito.when(mockHealthIndicator.health()).thenReturn(Health.up().build());
+    sutDelayedHealthIndicator.checkHealth();
+
+    // when
+    // backend health indicator goes down
+    Mockito.when(mockHealthIndicator.health()).thenReturn(Health.down().build());
+    sutDelayedHealthIndicator.checkHealth();
+
+    final Health actualHealth = sutDelayedHealthIndicator.health();
+
+    // then
+    // delayed health indicator is still up
+    Assertions.assertThat(actualHealth).isNotNull();
+    Assertions.assertThat(actualHealth.getStatus()).isEqualTo(Status.UP);
+  }
+
+  @Test
+  public void
+      shouldReportHealthStatusDownWhenBackendHealthIndicatorWasUpInThePastAndIsDownForMoreThanMaxDowntime()
+          throws InterruptedException {
+    // given
+    final var sutDelayedHealthIndicator =
+        new DelayedHealthIndicator(mockHealthIndicator, TEST_MAX_DOWNTIME);
+    // backend health indicator was up in the past
+    Mockito.when(mockHealthIndicator.health()).thenReturn(Health.up().build());
+    sutDelayedHealthIndicator.checkHealth();
+
+    // when
+    // backend health indicator goes down
+    Mockito.when(mockHealthIndicator.health()).thenReturn(Health.down().build());
+    sutDelayedHealthIndicator.checkHealth();
+
+    final Health actualHealthImmediate = sutDelayedHealthIndicator.health();
+
+    // wait for more then the configured max downtime
+    Thread.sleep(100);
```

Comment:
:see_no_evil: 

I don't like these kind of time based tests :sweat_smile: Could we test it differently? In our actor scheduler for example we can increase the clock, to test such things easily. I assume this is here not easily possible.

But we could just call the method twice to emulate such thing? Time based tests tend to be flaky quite often or take unnecessary long.

Reply:
In principle I agree with you. However, I think this can be a pragmatic exception. I reduced the time to 20 ms, so it won't take so long. I also ran them several times and they always gave me the same result.

I think passing in an `ActorClock`would be overkill. Also not sure which method you want to call twice to emulate the passing of time. 

I would like to keep it this way, to be honest.